freedomofpress/securedrop-client
# use latest semgrep

# Description

Use the latest version of semgrep in our dev environment and CI. This also fixes the recent semgrep error that started showing up 4 days ago, e.g.: https://circleci.com/gh/freedomofpress/securedrop-client/6169

After the next securedrop-client release, we plan to update almost all of our dev dependency specifications and pin to the latest dev-only packages.

# Test Plan
1. run `make semgrep` locally from `main` and confirm error:
```
semgrep error: invalid language: hcl
  --> https://semgrep.dev/...:7334
7333 |   languages:
7334 |   - hcl
7335 |   message: An EBS volume is configured without encryption enabled.
7336 |   patterns:

unsupported language: hcl. supported languages are: C#, JSON, Json, Kotlin, Rust, Yaml, c, cs, csharp, generic, go, golang, java, javascript, js, json, jsx, kotlin, kt, lua, ml, none, ocaml, php, py, python, python2, python3, rb, regex, rs, ruby, rust, ts, tsx, typescript, yaml
```
2. checkout this PR branch and build a new virtual environment with the latest semgrep (run `make venv`)
3. activate your new virtual environment and run `make semgrep` 
  - [ ] semgrep runs without error

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:


If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:


If these changes modify the database schema, you should include a database migration. Please check as applicable:
